numpy/core/src/common/simd/avx2/arithmetic.h

```diff
@@ -118,7 +119,16 @@
 #endif // !NPY_HAVE_FMA3
 
 // Horizontal add: Calculates the sum of all vector elements.
```

Comment:
please could you separate sum and sumup intrinsics for all SIMD extensions? something like that
```C
/***************************
 * Summation
 ***************************/
// reduce sum across vector
NPY_FINLINE npy_uint32 npyv_sum_u32(npyv_u32 a) {...}
NPY_FINLINE npy_uint64 npyv_sum_u64(npyv_u64 a)  {...}
NPY_FINLINE float  npyv_sum_f32(npyv_f32 a)  {...}
NPY_FINLINE double npyv_sum_f32(npyv_f64 a)  {...}

// extend sum across vector
NPY_FINLINE npy_uint16 npyv_sumup_u8(npyv_u8 a)  {...}
NPY_FINLINE npy_uint32 npyv_sumup_u16(npyv_u16 a)  {...}
```

Reply:
That's more clear.